src/workers/javascript/JavaScriptWorker.worker.ts

```diff
@@ -121,7 +120,8 @@ class JavaScriptWorker extends Backend {
         return ret;
     }
 
-    override _runCodeInternal(code: string): Promise<any> {
+    override runCode(extras: SyncExtras, code: string): Promise<any> {
+        this.extras = extras;
```

Comment:
I assume that input is still supported in JS. What about sleeping?

Reply:
Sleeping doesn't really seem to be a thing in JS. What exists is setTimeout, which is non-blocking. In order to properly support that, I think I would have to store the timers internally and await them before giving control back to the user. Our focus is currently on Python, so I don't plan on changing this yet.

Reply:
> What exists is setTimeout, which is non-blocking. In order to properly support that, I think I would have to store the timers internally and await them before giving control back to the user.

What exactly are you saying? Using setTimeout as usual doesn't work properly? Or do you mean making setTimeout blocking?

You could add a custom `sleep` function using `extras.syncSleep()`, which currently just means running the callback with the sleep event.

Reply:
What I meant is that a blocking sleep call does not exist in JavaScript by default. It can be emulated with Promises and awaiting them, but I haven't played around a lot with async/await in the JavaScript runner.
setTimeout is non-blocking and works, but these calls are not "awaited", so if you write eg 
```
setTimeout(() => console.log("Timer"), 5000)
```  
then the run will be finished immediately and the log statement will appear later in the actual console instead of the output field. This is because of the global overrides being done at the start and being cleared at the end of a run. That's why some awaiting mechanism should be added for these timeouts.